src/Processors/QueryPlan/FilterStep.cpp

```diff
@@ -50,6 +52,12 @@ FilterStep::FilterStep(
     , filter_column_name(std::move(filter_column_name_))
     , remove_filter_column(remove_filter_column_)
 {
+    for (const auto & output_node_ptr : actions_dag->getOutputs())
```

Comment:
But it is a filter, not an expression, how can I "apply" it to something? It will return either True or False.
What's going on here: inside updateDataHintsWithFilterActionsDAG I find INPUT columns (not ALIAS), and then try to build restrictions on such columns via <, >, <=, >= predicates with AND.